packages/EasyTest/src/Traits/ExceptionTrait.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyTest\Traits;
+
+use Closure;
+use RuntimeException;
+use Throwable;
+
+trait ExceptionTrait
+{
+    protected static bool $isInsideSafeCall = false;
+
+    protected ?Throwable $thrownException = null;
+
+    private bool $isThrownExceptionAssertionNeeded = false;
+
+    /**
+     * @param class-string<\Throwable>|null $previousException
+     */
+    protected function assertThrownException(
+        string $expectedException,
+        ?int $code = null,
+        ?string $previousException = null
+    ): void {
+        $this->isThrownExceptionAssertionNeeded = false;
+
+        self::assertNotNull($this->thrownException);
+
+        if ($this->thrownException instanceof $expectedException === false) {
+            throw $this->thrownException;
+        }
+
+        if (\str_starts_with($expectedException, 'App\\')) {
+            $message = "Didn't you forget to define the error code?";
+            self::assertNotNull($code, $message);
+            self::assertNotSame(0, $code, $message);
+        }
+
+        if ($code !== null) {
+            self::assertSame($code, $this->thrownException->getCode());
+        }
+
+        if ($previousException === null) {
+            self::assertNull($this->thrownException->getPrevious());
+        }
+
+        if ($previousException !== null) {
+            self::assertInstanceOf($previousException, $this->thrownException->getPrevious());
+        }
+    }
+
+    protected function safeCall(Closure $func): void
+    {
+        self::$isInsideSafeCall = true;
+
+        try {
+            $this->isThrownExceptionAssertionNeeded = true;
+            $func();
+        } catch (Throwable $exception) {
+            $this->thrownException = $exception;
+        }
+
+        self::$isInsideSafeCall = false;
+    }
+
+    /**
+     * This check is needed to make sure that the developer is not forgetting to assert the thrown exception,
+     * when using safeCall().
+     *
+     * This check should be called in the tearDown() method of the test class.
+     */
+    private function checkThrownExceptionAssertion(): void
+    {
+        if ($this->isThrownExceptionAssertionNeeded) {
+            throw new RuntimeException(
+                'ExceptionTrait::safeCall() must be followed by ExceptionTrait::assertThrownException()'
+            );
+        }
+    }
```

Comment:
In PBA projects we have a [`getErrorHandler` method](https://github.com/eonx-com/mastercard-pba-transact-api-v1/blob/release/src/tests/Util/ExceptionTrait.php) and we use it like so. Don't we need it here as well? 
![image](https://user-images.githubusercontent.com/20467102/216509918-a6284682-00e2-4c59-ac2c-a41920746fc7.png)

Reply:
This will be implemented in another way - new stub with static methods.